src/Stratis.FederatedPeg.Features.FederationGateway/Notifications/BlockObserver.cs

```diff
@@ -64,11 +71,27 @@ protected override void OnNextCore(ChainedHeaderBlock chainedHeaderBlock)
         {
             this.walletSyncManager.ProcessBlock(chainedHeaderBlock.Block);
 
-            this.federationGatewayClient.PushCurrentBlockTipAsync(
-                new BlockTipModel(
-                    chainedHeaderBlock.ChainedHeader.HashBlock,
-                    chainedHeaderBlock.ChainedHeader.Height,
-                    (int)this.depositExtractor.MinimumDepositConfirmations)).ConfigureAwait(false).GetAwaiter().GetResult();
+            // Cancel previous sending to avoid first sending new tip and then sending older tip.
+            if ((this.pushBlockTipTask != null) && !this.pushBlockTipTask.IsCompleted)
+            {
+                this.cancellationSource.Cancel();
+                this.pushBlockTipTask.GetAwaiter().GetResult();
+
+                this.pushBlockTipTask = null;
+                this.cancellationSource = null;
+            }
+
+            var blockTipModel = new BlockTipModel(chainedHeaderBlock.ChainedHeader.HashBlock,chainedHeaderBlock.ChainedHeader.Height, (int)this.depositExtractor.MinimumDepositConfirmations);
+
+            // There is no reason to wait for the message to be sent.
+            // Awaiting REST API call will only slow this callback.
+            // Callbacks never supposed to do any IO calls or web requests.
+            // Instead we start sending the message and if next block was connected faster than the message was sent we
+            // are canceling it and sending the next tip.
+            // Receiver of this message doesn't care if we are not providing tips for every block we connect,
+            // it just requires to know about out latest state.
+            this.cancellationSource = new CancellationTokenSource();
+            this.pushBlockTipTask = Task.Run(async () => await this.federationGatewayClient.PushCurrentBlockTipAsync(blockTipModel, this.cancellationSource.Token).ConfigureAwait(false));

```

Comment:
this.pushBlockTipTask = this.federationGatewayClient.PushCurrentBlockTipAsync(blockTipModel, this.cancellationSource.Token);
```suggestion
            this.pushBlockTipTask = this.federationGatewayClient.PushCurrentBlockTipAsync(blockTipModel, this.cancellationSource.Token);
```

Reply:
There is a difference between those 2. 

What you suggest will execute till `await this.policy.ExecuteAsync(`. 

What there is will wrap call in a task and will immediately continue executing next line